common/functions/src/scalars/semi_structureds/check_json.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+use std::sync::Arc;
+
+use common_arrow::arrow::bitmap::Bitmap;
+use common_datavalues::prelude::*;
+use common_datavalues::with_match_physical_primitive_type;
+use common_exception::ErrorCode;
+use common_exception::Result;
+use serde_json::Value as JsonValue;
+
+use crate::scalars::Function;
+use crate::scalars::FunctionDescription;
+use crate::scalars::FunctionFeatures;
+
+#[derive(Clone)]
+pub struct CheckJsonFunction {
+    display_name: String,
+}
+
+impl CheckJsonFunction {
+    pub fn try_create(display_name: &str) -> Result<Box<dyn Function>> {
+        Ok(Box::new(CheckJsonFunction {
+            display_name: display_name.to_string(),
+        }))
+    }
+
+    pub fn desc() -> FunctionDescription {
+        FunctionDescription::creator(Box::new(Self::try_create)).features(
+            FunctionFeatures::default()
+                .deterministic()
+                .monotonicity()
+                .num_arguments(1),
+        )
+    }
+}
+
+impl Function for CheckJsonFunction {
+    fn name(&self) -> &str {
+        &*self.display_name
+    }
+
+    fn return_type(&self, args: &[&DataTypePtr]) -> Result<DataTypePtr> {
+        if args[0].data_type_id() == TypeID::Null {
+            return Ok(NullType::arc());
+        }
+
+        return Ok(Arc::new(NullableType::create(StringType::arc())));
+    }
+
+    fn eval(&self, columns: &ColumnsWithField, input_rows: usize) -> Result<ColumnRef> {
+        let mut column = columns[0].column();
+        let mut _all_null = false;
+        let mut source_valids: Option<&Bitmap> = None;
+        if column.is_nullable() {
+            (_all_null, source_valids) = column.validity();
+            let nullable_column: &NullableColumn = Series::check_get(column)?;
+            column = nullable_column.inner();
+        }
+
+        if column.data_type_id() == TypeID::Null {
+            return NullType::arc().create_constant_column(&DataValue::Null, input_rows);
+        }
+
+        let mut builder = NullableColumnBuilder::<Vu8>::with_capacity(input_rows);
+
+        with_match_physical_primitive_type!(column.data_type_id().to_physical_type(), |$T| {
```

Comment:
We should use `with_match_primitive_type_id` here, as both `Uint32` and `Date32` have some physical type.
Use `columns[0].field().data_type_id()` instead of `column.data_type_id()` for the same reason.